components/core/tests/test-ir_encoding_methods.cpp

```diff
@@ -357,6 +373,30 @@ auto count_num_leaves(nlohmann::json const& root) -> size_t {
 
     return num_leaves;
 }
+
+template <typename encoded_variable_t>
+auto unpack_and_assert_serialization_failure(
+        std::stringstream& buffer,
+        Serializer<encoded_variable_t>& serializer
+) -> bool {
+    string msgpack_bytes{buffer.str()};
+    buffer.str({});
+    buffer.clear();
+    auto const msgpack_obj_handle{msgpack::unpack(msgpack_bytes.data(), msgpack_bytes.size())};
+    auto const msgpack_obj{msgpack_obj_handle.get()};
+    if (msgpack::type::MAP != msgpack_obj.type) {
+        return false;
+    }
+    if (serializer.serialize_msgpack_map(msgpack_obj.via.map)) {
+        // Serialization should fail
+        return false;
+    }
+    if (false == serializer.get_ir_buf_view().empty()) {
```

Comment:
Can we add a Catch2 assertion to check that the buffer that's passed in is empty?